File: functions_environment/example_read_env.py
# These lines need to be added at the beggining of the codes to automatically set the environment
import json
import os
from detect_environment import detect_environment
from load_conf import load_conf

# Automatically detect environment
env = detect_environment()

# Load configuration
config = load_conf(env)

# Access input and output directories
input_dir = config['input_dir']
output_dir = config['output_dir']

print(f'Detected environment: {env}')
print(f'Using input directory: {input_dir}')
print(f'Using output directory: {output_dir}')

# Your main code here
# For example:
# process_data(input_dir, output_dir)

File: functions_environment/example_read_env.m
% Main script
addpath(genpath([pwd,filesep,'functions_environment']))

% Automatically detect environment
env = detect_environment();

% Load configuration
config = load_conf(env);

% Access input and output directories
input_dir = config.input_dir;
output_dir = config.output_dir;

fprintf('Detected environment: %s\n', env);
fprintf('Using input directory: %s\n', input_dir);
fprintf('Using output directory: %s\n', output_dir);

% Your main code here